src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/AbstractSearchAction.java

```diff
@@ -68,15 +68,15 @@
 
     private final String index;
     private final Class<T> clazz;
-    private final String urlPath;
+    private final List<String> urlPaths;
     private final ActionType<SearchResponse> actionType;
 
     private final Logger logger = LogManager.getLogger(AbstractSearchAction.class);
 
-    public AbstractSearchAction(String urlPath, String index, Class<T> clazz, ActionType<SearchResponse> actionType) {
+    public AbstractSearchAction(List<String> urlPaths, String index, Class<T> clazz, ActionType<SearchResponse> actionType) {
```

Comment:
Add a constructor that just takes `String` now to avoid future callers adding new routes to have to pass a list.

Reply:
Sure, I like that!

Reply:
After looking at the updated changes, I think future callers can use a list but pass in one item if needed. What do you think?